v2/internal/controllers/edge_case_test.go

```diff
@@ -221,3 +176,69 @@ func Test_CreateStorageAccountWithoutRequiredProperties_Rejected(t *testing.T) {
 	// Create it again
 	tc.CreateResourcesAndWait(acctCopy)
 }
+
+func Test_AzureNameImmutability(t *testing.T) {
```

Comment:
minor: Name of the test should include what it's testing. Possibly `Test_AzureName_IsImmutableOnceSuccessfullyCreated`?

Reply:
Same below for the owner one.

Reply:
You should also cover the cases I mentioned in Teams where the object isn't successfully created yet. We should allow those mutations if we can.

Reply:
Done!